column_buffer.go

```diff
@@ -716,46 +718,73 @@ func (col *booleanColumnBuffer) Pages() Pages { return onePage(col.Page()) }
 
 func (col *booleanColumnBuffer) Page() BufferedPage { return &col.booleanPage }
 
-func (col *booleanColumnBuffer) Reset() { col.values = col.values[:0] }
+func (col *booleanColumnBuffer) Reset() {
+	col.values = col.values[:0]
+	col.offset = 0
+	col.numValues = 0
+}
 
-func (col *booleanColumnBuffer) Cap() int { return cap(col.values) }
+func (col *booleanColumnBuffer) Cap() int { return 8 * cap(col.values) }
 
-func (col *booleanColumnBuffer) Len() int { return len(col.values) }
+func (col *booleanColumnBuffer) Len() int { return int(col.numValues) }
 
 func (col *booleanColumnBuffer) Less(i, j int) bool {
-	return col.values[i] != col.values[j] && !col.values[i]
+	a := col.valueAt(i)
+	b := col.valueAt(j)
+	return a != b && !a
 }
 
-func (col *booleanColumnBuffer) Swap(i, j int) {
-	col.values[i], col.values[j] = col.values[j], col.values[i]
+func (col *booleanColumnBuffer) valueAt(i int) bool {
+	j := uint32(i) / 8
+	k := uint32(i) % 8
+	return ((col.values[j] >> k) & 1) != 0
 }
 
-func (col *booleanColumnBuffer) Write(b []byte) (int, error) {
-	return col.WriteBooleans(bits.BytesToBool(b))
+func (col *booleanColumnBuffer) setValueAt(i int, v bool) {
```

Comment:
Thread safety issue?

Reply:
The ColumnBuffer types are not intended to be used concurrently from multiple goroutines so I believe we're ok here, unless there was something else you wanted to point out?